roofit/roofitcore/inc/RooFit/Detail/EvaluateFuncs.h

```diff
@@ -97,6 +97,74 @@ inline double poissonEvaluate(double x, double par)
    }
 }
 
+inline double polyInterpValue(unsigned int i, double *polCoeff, double x)
+{
+   unsigned int offset = 6 * i;
+
+   double a = polCoeff[offset + 0];
+   double b = polCoeff[offset + 1];
+   double c = polCoeff[offset + 2];
+   double d = polCoeff[offset + 3];
+   double e = polCoeff[offset + 4];
+   double f = polCoeff[offset + 5];
+
+   // evaluate the 6-th degree polynomial using Horner's method
+   return 1. + x * (a + x * (b + x * (c + x * (d + x * (e + x * f)))));
+}
+
+inline double flexibleInterpVarProcessParam(unsigned int code, unsigned int i, double *polCoeff, double low,
+                                            double high, double boundary, double nominal, double paramVal, double total)
+{
+   if (code == 0) {
+      // piece-wise linear
+      if (paramVal > 0)
+         return total + paramVal * (high - nominal);
+      else
+         return total + paramVal * (nominal - low);
+   } else if (code == 1) {
+      // pice-wise log
+      if (paramVal >= 0)
+         return total * pow(high / nominal, +paramVal);
+      else
+         return total * pow(low / nominal, -paramVal);
+   } else if (code == 2) {
+      // parabolic with linear
+      double a = 0.5 * (high + low) - nominal;
+      double b = 0.5 * (high - low);
+      double c = 0;
+      if (paramVal > 1) {
+         return total + (2 * a + b) * (paramVal - 1) + high - nominal;
+      } else if (paramVal < -1) {
+         return total + -1 * (2 * a - b) * (paramVal + 1) + low - nominal;
+      } else {
+         return total + a * pow(paramVal, 2) + b * paramVal + c;
+      }
+   } else if (code == 3) {
+      // parabolic version of log-normal
+      double a = 0.5 * (high + low) - nominal;
+      double b = 0.5 * (high - low);
+      double c = 0;
+      if (paramVal > 1) {
+         return total + (2 * a + b) * (paramVal - 1) + high - nominal;
+      } else if (paramVal < -1) {
+         return total + -1 * (2 * a - b) * (paramVal + 1) + low - nominal;
+      } else {
+         return total + a * pow(paramVal, 2) + b * paramVal + c;
+      }
+   } else if (code == 4) {
+      double x = paramVal;
+      if (x >= boundary) {
+         return total * std::pow(high / nominal, +paramVal);
+      } else if (x <= -boundary) {
+         return total * std::pow(low / nominal, -paramVal);
+      } else if (x != 0) {
+         return total * polyInterpValue(i, polCoeff, x);
+      }
+   }
+
+   return 0.0;
```

Comment:
```suggestion
   return total;
```
To fix the test failures and the problem that we sometimes get NaN